Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/parser/TestPatientDiscoveryMessageHelper.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    private static JAXBElement<PRPAMT201306UV02QueryByParameter> createQueryParams(String firstName, String lastName,
+        String gender, String birthTime, II subjectId) {
+
+        II id = new II();
+        id.setRoot(QUERY_ID_ROOT);
+        id.setExtension(QUERY_ID_EXTENSION);
+        CS statusCode = new CS();
+        statusCode.setCode("new");
+
+        PRPAMT201306UV02QueryByParameter params = new PRPAMT201306UV02QueryByParameter();
+        params.setQueryId(id);
+        params.setStatusCode(statusCode);
+        params.setParameterList(createParamList(firstName, lastName, gender, birthTime, subjectId));
+
+        return new JAXBElement<>(new QName("urn:hl7-org:v3", "queryByParameter"),
+            PRPAMT201306UV02QueryByParameter.class, params);
+    }
+
+    private static PRPAMT201306UV02ParameterList createParamList(String firstName, String lastName, String gender,
+        String birthTime, II subjectId) {
+
+        PRPAMT201306UV02ParameterList paramList = new PRPAMT201306UV02ParameterList();
+        paramList.getLivingSubjectAdministrativeGender().add(createGender(gender));
+        paramList.getLivingSubjectBirthTime().add(createBirthTime(birthTime));
+        paramList.getLivingSubjectName().add(createName(firstName, lastName));
+        paramList.getLivingSubjectId().add(createSubjectId(subjectId));
+        return paramList;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectId createSubjectId(II subjectId) {
+        PRPAMT201306UV02LivingSubjectId id = new PRPAMT201306UV02LivingSubjectId();
+        if (subjectId != null) {
+            id.getValue().add(subjectId);
+        }
+
+        return id;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectName createName(String firstName, String lastName) {
+        org.hl7.v3.ObjectFactory factory = new org.hl7.v3.ObjectFactory();
+        ENExplicit name = factory.createENExplicit();
+
+        if (lastName != null && lastName.length() > 0) {
+            EnExplicitFamily familyName = new EnExplicitFamily();
+            familyName.setPartType("FAM");
+            familyName.setContent(lastName);
+
+            name.getContent().add(factory.createENExplicitFamily(familyName));
+        }
+
+        if (firstName != null && firstName.length() > 0) {
+            EnExplicitGiven givenName = new EnExplicitGiven();
+            givenName.setPartType("GIV");
+            givenName.setContent(firstName);
+
+            name.getContent().add(factory.createENExplicitGiven(givenName));
+        }
+
+        PRPAMT201306UV02LivingSubjectName subjectName = new PRPAMT201306UV02LivingSubjectName();
+        subjectName.getValue().add(name);
+        return subjectName;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectBirthTime createBirthTime(String birthTime) {
+        PRPAMT201306UV02LivingSubjectBirthTime subjectBirthTime = new PRPAMT201306UV02LivingSubjectBirthTime();
+        IVLTSExplicit bday = new IVLTSExplicit();
```

Comment:
its a private method and the calling method always passes a constant when createBirthTime() is called